lib/Elevate/Components/Softaculous.pm

```diff
@@ -0,0 +1,75 @@
+package Elevate::Components::Softaculous;
+
+=encoding utf-8
+
+=head1 NAME
+
+Elevate::Components::Softaculous
+
+=head2 pre_distro_upgrade
+
+If we can query the version of Softaculous through their CLI tool, it is installed.
+
+=head2 post_distro_upgrade
+
+If Softaculous is installed, re-install it after the upgrade.
+
+NOTE: This needs to happen after cPanel is updated to work with the new OS, since Softaculous relies on cPanel PHP.
+
+=cut
+
+use cPstrict;
+
+use Elevate::Fetch     ();
+use Elevate::StageFile ();
+
+use Cpanel::Binaries        ();
+use Cpanel::SafeRun::Object ();
+
+use Log::Log4perl qw(:easy);
+
+use parent qw{Elevate::Components::Base};
+
+sub pre_distro_upgrade ($self) {
+
+    my $SOFTACULOUS_CLI_PATH = '/usr/local/cpanel/whostmgr/docroot/cgi/softaculous/cli.php';
+
+    return unless -r $SOFTACULOUS_CLI_PATH;
+
+    my $sr = Cpanel::SafeRun::Object->new(
+        program => Cpanel::Binaries::path('php'),
+        args    => [ $SOFTACULOUS_CLI_PATH, '--version' ],
+    );
+
+    return if $sr->exec_failed() || $sr->to_exception();
+
+    my $version = $sr->stdout();
+    chomp $version;
+
+    if ($version) {
+        INFO('Softaculous has been detected. The system will re-install that software after the distro upgrade.');
+        Elevate::StageFile::update_stage_file( { softaculous => $version } );
+    }
+
+    return;
+}
+
+sub post_distro_upgrade ($self) {
+
+    my $version = Elevate::StageFile::read_stage_file( 'softaculous', '' );
+    return unless $version;
+
+    my $path = Elevate::Fetch::script( 'https://files.softaculous.com/install.sh', 'softaculous_install' );
+
+    if ($path) {
+        INFO('Re-installing Softaculous:');
+        LOGDIE('Re-installation of Softaculous failed.') if !$self->ssystem( $path, '--reinstall' );
```

Comment:
I don't think that this failing should be fatal for the script.  Can we provide the customer with some information on how to reinstall softaculous to get it working again and add this to the final notifications at the end instead?

Reply:
I went back and forth about whether it should be fatal or not. Changing it back.

Reply:
Is it a problem they can clean up after Elevate completes? Continue. If it's going to break the rest of elevate then stop and make them fix it.

Reply:
> Is it a problem they can clean up after Elevate completes?

Yes, this is what customers are having to do now.  The released version of Elevate completes successfully now without any of this in place.